### PR TITLE
fix: `addr_space_max_bits` in `VolatileBoundaryChip` constructor

### PR DESCRIPTION
**Pull request by jonathanpwang**
We were using `as_height` but it is actually `log2(1 + 2^as_height)`.

### CODE REVIEW

File: crates/vm/src/system/memory/controller/mod.rs [hunk 1]
```diff
@@ -232,13 +232,19 @@ impl<F: PrimeField32> MemoryController<F> {
     ) -> Self {
         let range_checker_bus = range_checker.bus();
         let initial_memory = AddressMap::from_mem_config(&mem_config);
+        assert!(mem_config.pointer_max_bits <= F::bits() - 2);
+        assert!(mem_config.as_height <= F::bits() - 2);
+        let addr_space_max_bits = log2_strict_usize(
+            (mem_config.as_offset + 2u32.pow(mem_config.as_height as u32)).next_power_of_two()
```
Comment (zlangley) [inline crates/vm/src/system/memory/controller/mod.rs:238]: nit: You can use `log2_ceil_usize` to skip the `next_power_of_two()`.

File: crates/vm/tests/integration_test.rs [hunk 2]
```diff
@@ -56,7 +57,7 @@ where
 
 fn test_native_config() -> NativeConfig {
     NativeConfig {
-        system: SystemConfig::default(),
+        system: SystemConfig::new(3, MemoryConfig::new(2, 1, 16, 29, 15, 32, 1024), 0),
```
Comment (zlangley) [inline crates/vm/tests/integration_test.rs:60]: Good to know this config works now, but I'm not sure it's better than the old one for testing since native address space is 4.

**Review by zlangley — APPROVED** [review]

File: crates/vm/src/system/memory/controller/mod.rs (hunk 1 above)
Comment (jonathanpwang) [reply crates/vm/src/system/memory/controller/mod.rs:238]: thanks, forgot we had this

File: crates/vm/tests/integration_test.rs (hunk 2 above)
Comment (jonathanpwang) [reply crates/vm/tests/integration_test.rs:60]: 4 is still supported right? it exactly hits the edge case: 4 < 1 + 2^2

Comment (zlangley) [reply crates/vm/tests/integration_test.rs:60]: Ah, right.

File: crates/vm/src/system/memory/controller/mod.rs (hunk 1 above)
Comment (zlangley) [reply crates/vm/src/system/memory/controller/mod.rs:238]: Did you mean to change this?

Comment (jonathanpwang) [reply crates/vm/src/system/memory/controller/mod.rs:238]: hm some weird failed save
